tests/5.0/teams_loop/test_target_teams_loop_defaultmap.c

```diff
@@ -0,0 +1,249 @@
+    host_data[i] = 0;
+  }
+#pragma omp target teams loop defaultmap(from)
+  for (int i = 0; i < N; i++) {
+    host_data[i] = i;
+  }
+  // checking results
+  for (int i = 0; i < N; i++) {
+    OMPVV_TEST_AND_SET(errors, host_data[i] != i);
+  }
+  return errors;
+}
+
+/**
+  This is a basic test to demonstrate defaultmap(to)
+  clause with target teams loop construct.
+*/
+int testDefaultMapTo() {
+  int errors = 0;
+  int device_data[N], host_data[N];
+  for (int i = 0; i < N; i++) {
+    device_data[i] = i;
+    host_data[i] = 0;
+  }
+#pragma omp target teams loop defaultmap(to) map(tofrom: host_data)
+  for (int i = 0; i < N; i++) {
+    host_data[i] += device_data[i];
+  }
+  // checking results
+  for (int i = 0; i < N; i++) {
+    OMPVV_TEST_AND_SET(errors, host_data[i] != i);
+  }
+  return errors;
+}
+
+/**
+  This is a basic test to demonstrate defaultmap(default)
+  clause with target teams loop construct.
+*/
+int testDefaultMapDefault() {
+  int errors = 0;
+  int device_data[N], host_data[N];
+  for (int i = 0; i < N; i++) {
+    device_data[i] = i;
+    host_data[i] = 0;
+  }
+#pragma omp target teams loop defaultmap(default)
+  for (int i = 0; i < N; i++) {
+    host_data[i] += device_data[i];
+    device_data[i] = 2*i;
+  }
+  // checking results
+  for (int i = 0; i < N; i++) {
+    OMPVV_TEST_AND_SET(errors, host_data[i] != i);
+    OMPVV_TEST_AND_SET(errors, device_data[i] != 2*i);
+  }
+  return errors;
+}
+
+/**
+  This is a basic test to demonstrate defaultmap(firstprivate)
+  clause with target teams loop construct.
+*/
+int testDefaultMapFirstPrivate() {
+  int errors = 0;
+  int device_data[N], host_data[N];
+  for (int i = 0; i < N; i++) {
+    device_data[i] = i;
+    host_data[i] = 0;
+  }
+#pragma omp target teams loop defaultmap(firstprivate)
+  for (int i = 0; i < N; i++) {
+    host_data[i] = device_data[i];
+  }
+  // checking results
+  for (int i = 0; i < N; i++) {
+    OMPVV_TEST_AND_SET(errors, host_data[i] != 0);
+  }
+  return errors;
+}
+
+/**
+  This is a basic test to demonstrate defaultmap(tofrom:aggregate)
+  clause with target teams loop construct.
+*/
+int testDefaultMapToFromAggr() {
+  int errors = 0;
+  int device_data[N], host_data[N];
+  for (int i = 0; i < N; i++) {
+    device_data[i] = i;
+    host_data[i] = 0;
+  }
+#pragma omp target teams loop defaultmap(tofrom:aggregate) map(from: host_data)
+  for (int i = 0; i < N; i++) {
+    host_data[i] += device_data[i];
+    device_data[i] = 2*i;
+  }
+  // checking results
+  for (int i = 0; i < N; i++) {
+    OMPVV_TEST_AND_SET(errors, host_data[i] != i);
+    OMPVV_TEST_AND_SET(errors, device_data[i] != 2*i);
+  }
+  return errors;
+}
+
+/**
+  This is a basic test to demonstrate defaultmap(tofrom:scalar)
+  clause with target teams loop construct.
+*/
+int testDefaultMapToFromScalar() {
+  int errors = 0;
+  int x = 0;
+#pragma omp target teams loop defaultmap(tofrom:scalar)
+  for (int i = 0; i < N; i++) {
+    x = N;
```

Comment:
"x" is shared across the teams and there is a race to update it. It may not be apparent as all threads are setting it ti N, but this leads to a non-conformant OpenMP program.
 To check the 'tofrom' consider: 
#pragma omp target teams loop defaultmap(tofrom:scalar) map(from: array) 
for (int i = 0; i < N; i++) {
#pragma omp atomic
    x += i; //this checks if the 'to' for x is working correctly
array[i] = x + 2;
}
//Then check array and x values on host